homeassistant/components/brother/config_flow.py

```diff
@@ -127,6 +140,64 @@ async def async_step_zeroconf_confirm(
             },
         )
 
+    async def async_step_reconfigure(
+        self, _: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a reconfiguration flow initialized by the user."""
+        entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+
+        if TYPE_CHECKING:
+            assert entry is not None
+
+        self.entry = entry
+
+        return await self.async_step_reconfigure_confirm()
+
+    async def async_step_reconfigure_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a reconfiguration flow initialized by the user."""
+        errors = {}
+
+        if TYPE_CHECKING:
+            assert self.entry is not None
+
+        if user_input is not None:
+            try:
+                await validate_input(self.hass, user_input, self.entry.unique_id)
+            except InvalidHost:
+                errors[CONF_HOST] = "wrong_host"
+            except (ConnectionError, TimeoutError):
+                errors["base"] = "cannot_connect"
+            except SnmpError:
+                errors["base"] = "snmp_error"
+            except AnotherDevice:
+                errors["base"] = "another_device"
+            else:
+                self.hass.config_entries.async_update_entry(
+                    self.entry,
+                    data={
+                        CONF_HOST: user_input[CONF_HOST],
+                        CONF_TYPE: self.entry.data[CONF_TYPE],
+                    },
```

Comment:
Maybe this is better to not risk accidentally discarding data in the future:
```suggestion
                    data=self.entry.data | {CONF_HOST: user_input[CONF_HOST]},
```

Reply:
Good point!